url-plugins/src/main/java/co/cask/hydrator/plugin/realtime/config/UrlFetchRealtimeSourceConfig.java

```diff
@@ -0,0 +1,179 @@
+
+
+  public UrlFetchRealtimeSourceConfig(String url,
+                                      long intervalInSeconds) {
+    this.url = url;
+    this.intervalInSeconds = intervalInSeconds;
+  }
+
+  public UrlFetchRealtimeSourceConfig(String url,
+                                      long intervalInSeconds,
+                                      String requestHeaders) {
+    this.url = url;
+    this.intervalInSeconds = intervalInSeconds;
+    this.requestHeadersString = requestHeaders;
+  }
+
+  public String getUrl() {
+    return url;
+  }
+
+  public long getIntervalInSeconds() {
+    return intervalInSeconds;
+  }
+
+  public boolean hasCustomRequestHeaders() {
+    return !Strings.isNullOrEmpty(requestHeadersString);
+  }
+
+  public Map<String, String> getRequestHeadersMap() {
+    return convertHeadersToMap(requestHeadersString);
+  }
+
+  public boolean shouldFollowRedirects() {
+    return followRedirects;
+  }
+
+  public Charset getCharset() {
+    return (Strings.isNullOrEmpty(charsetString)) ? DEFAULT_CHARSET : Charset.forName(charsetString);
+  }
+
+  public int getConnectTimeout() {
+    return connectTimeout;
+  }
+
+  public int getReadTimeout() {
+    return readTimeout;
+  }
+
+  @SuppressWarnings("ConstantConditions")
+  public void validate() {
+    try {
+      new URL(url);
+    } catch (MalformedURLException e) {
+      throw new IllegalArgumentException(String.format("URL '%s' is malformed: %s", url, e.getMessage()), e);
+    }
+    if (intervalInSeconds <= 0) {
+      throw new IllegalArgumentException(String.format(
+        "Invalid intervalInSeconds %d. Interval must be greater than 0.", intervalInSeconds));
+    }
+    if (connectTimeout < 0) {
+      throw new IllegalArgumentException(String.format(
+        "Invalid connectTimeout %d. Timeout must be a positive number.", connectTimeout));
+    }
+    if (readTimeout < 0) {
+      throw new IllegalArgumentException(String.format(
+        "Invalid readTimeout %d. Interval must be a positive number.", readTimeout));
+    }
+    try {
+      if (!Strings.isNullOrEmpty(charsetString)) {
+        Charset.forName(charsetString);
+      }
+    } catch (UnsupportedCharsetException e) {
+      throw new IllegalArgumentException(String.format("Invalid charset string %s.", charsetString));
+    }
+    try {
+      convertHeadersToMap(requestHeadersString);
+    } catch (Exception e) {
+      throw new IllegalArgumentException(String.format("Could not parse requestHeadersString string due to: '%s'.",
+                                                       e.getMessage()));
+    }
+  }
+
+  private Map<String, String> convertHeadersToMap(String headersString) {
+    Map<String, String> headersMap = new HashMap<>();
+    if (!Strings.isNullOrEmpty(headersString)) {
+      List<String> headerChunks = Arrays.asList(headersString.split(DELIMITER));
+      for (String chunk : headerChunks) {
+        List<String> keyValueList = Arrays.asList(chunk.split(KV_DELIMITER));
```

Comment:
doesn't matter since this is only called once, but for future reference, split needs to compile the regex, so it's faster to store Pattern.compile(regex) once, then do pattern.split(str, n) each time you need to split.